Packs/RiskIQDigitalFootprint/Integrations/RiskIQDigitalFootprint/RiskIQDigitalFootprint.py

```diff
@@ -0,0 +1,2451 @@
+        if max_records <= 0 or max_records > 10000:
+            raise ValueError(MESSAGES['MAX_RECORDS_VALIDATION'])
+    except ValueError:
+        raise ValueError(MESSAGES['MAX_RECORDS_VALIDATION'])
+
+    return request_timeout, max_records
+
+
+def nested_to_flat(src: Dict[str, Any], key: str) -> Dict[str, Any]:
+    """
+    Convert nested dictionary to flat by contact the keys. Also converts keys in pascal string format.
+
+    :param src: sub-dictionary that needs to convert from nested to flat. (e.g. "foo": {"bar": "some-value"})
+    :param key: main key of sub-dictionary (e.g "foo")
+    :return: flat dictionary with pascal formatted keys (e.g. {"FooBar": "some-value"})
+    """
+
+    flat_dict: Dict[str, str] = {}
+    for sub_key, sub_value in src.items():
+        parent_key = key
+        if ' ' in sub_key:
+            sub_key = ''.join(sub_key.split(' '))
+        pascal_key = '{}{}'.format(parent_key, sub_key[0].upper() + sub_key[1:])
+        if parent_key in sub_key:
+            pascal_key = sub_key
+        flat_dict[pascal_key] = sub_value
+
+    return flat_dict
+
+
+def prepare_context_dict(response_dict: Dict[str, Any],
+                         keys_with_hierarchy: Union[tuple, str] = (),
+                         exclude_keys: Union[tuple, str] = ()) -> Dict[str, str]:
+    """
+    Prepare the context dictionary as per the standards.
+
+    :param response_dict: dictionary getting from API response that contains sub-dictionaries
+    :param keys_with_hierarchy: list of keys that contains sub-dictionary as its value.
+    :param exclude_keys: keys need to exclude.
+    :return: single level dictionary
+    """
+    simple_dict: Dict[str, str] = {}
+    for key, value in response_dict.items():
+        if key in keys_with_hierarchy:
+            simple_dict.update(nested_to_flat(response_dict.get(key, {}), key))
+        elif key not in exclude_keys:
+            simple_dict[key] = value
+    return simple_dict
+
+
+def validate_asset_connections_args(type_arg: str, global_arg: str) -> Tuple[str, Any]:
+    """
+    Validate the arguments passed by the user
+
+    :param type_arg: asset_type
+    :param global_arg: argument which defines if connected assets are to be fetched from global inventory
+    :return: type_arg, global_arg: asset_type and global arguments passed by user or message describing error that
+     occurred in parameter validation
+     occurred during validation
+    """
+    if type_arg not in VALID_ASSET_TYPES:
+        raise ValueError(MESSAGES['INVALID_ASSET_TYPE_ERROR'])
+    if global_arg:
+        global_arg = global_arg.lower()
+        try:
+            global_arg = argToBoolean(global_arg)
+        except ValueError:
+            raise ValueError(MESSAGES['INVALID_BOOLEAN_VALUE_ERROR'].format('global'))
+
+    if type_arg == 'ASN':
+        type_arg = 'AS'
+
+    return type_arg, global_arg
+
+
+def get_comma_separated_values(values: List[Dict[str, Any]], key: str) -> str:
```

Comment:
I think the naming of this function could be improved upon.

Reply:
@avidan-H Sure, we will try to improve the naming of this function.